docs: update readme (#14)
BedrockStreaming/prescaling-exporter: README.md

> The project includes an exporter that calculates the metric to provide according to the annotations defined on the hpa. and an API that allows to declare an event and a capacity multiplier

Reviewer: ```suggestion
The exporter exposes a metric calculated according to the annotations defined on the HPA. The project also exposes an API that allows to register an event with a capacity multiplier.
```

## Requirements

   - Prometheus Stack or Victoria Metrics Stack
   - [Prometheus Adapater](https://github.com/kubernetes-sigs/prometheus-adapter) 

> *Info : It is quite possible to use this solution with another observability stack than Prometheus. For example Datadog or Newrelic, but we don't have a configuration example.* 

Reviewer: ```suggestion
*Info: It is quite possible to use this solution with another observability stack than Prometheus. For example, Datadog or Newrelic, but we do not provide a configuration example.* 
```

# Install
## Kubernetes Deployement

- Clone repo
- Run this command with Helm3

```bash
helm install prescaling-exporter ./helm/prescaling-exporter -n prescaling-exporter --create-namespace
```

> *You can use skaffold if you want.* 

Reviewer: For a note, I'd rather use `>`:
```suggestion
> You can use skaffold if you want.
```

> ## Configure a Horizontal Pod Autoscaler

Reviewer: ```suggestion
## Configure an Horizontal Pod Autoscaler
```

> To be able to pre-scale an application every day before a traffic spike, the only thing to do is to add the

Reviewer: ```suggestion
To be able to pre-scale an application every day before a traffic spike, you must add the
```

annotations.scaling.exporter.replica.min  | "integer"

```
apiVersion: autoscaling/v2beta1
kind: HorizontalPodAutoscaler
metadata:
  name: "{{ .Release.Name }}"
  annotations:
    annotations.scaling.exporter.replica.min: "{{ .Values.hpa.annotations.replica_min"
    annotations.scaling.exporter.time.end: "{{ .Values.hpa.annotations.time_end }}"
    annotations.scaling.exporter.time.start: "{{ .Values.hpa.annotations.time_start }}"
spec:
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: "{{ .Release.Name }}"
  minReplicas: {{ .Values.hpa.minReplicas }}
  maxReplicas: {{ .Values.hpa.maxReplicas }}
  metrics:
  - type: External
    external:
      metricName: "prescale_metric"
      metricSelector:
          matchLabels:
            deployment: "{{ .Release.Name }}"
      targetValue: 10
```

> *it's important to set the target to 10, the value provided for scale is 11, the addition of pod will be gradual. Currently, the increment is carried out 10% by 10%.*

Reviewer: ```suggestion
> It's important to set the `targetValue` to 10. The metric's value provided by the exporter in order to scale is 11. The scale up of pods will be gradual. Currently, the increment is carried out 10% by 10%.
```

## Configure prometheus adapter

> A configuration example using the prometheus adapter to supply the metric to the Kubernetes cluster. 

Reviewer: ```suggestion
Here is a configuration example using the prometheus adapter to supply the metric to the Kubernetes cluster:
```

```
  - "metricsQuery": "avg(<<.Series>>{<<.LabelMatchers>>})"
    "name":
      "as": "prescale_metric"
    "resources":
      "overrides":
        "namespace":
          "resource": "namespace"
    "seriesQuery": "prescale_metric"
```

## Configure parameters 

> You can modify the application's settings to use different annotations or a different port. Configurable variable environments in the chart values are used to do this.

Reviewer: ```suggestion
You can modify the application's settings to use different annotations or a different port. Configurable variable environments in the chart values are used to do this:
```

Parameters            | Default values                            | Comment
---                   | ---                                       | --- 

> Namespace             | "prescaling-exporter"                     | Namespace for PrescalingEvent

Reviewer: Why not set the namespace to `prescaling` by default? `exporter` is tightly coupled to Prometheus IMHO.

```suggestion
Namespace             | "prescaling"                     | Namespace for the prescaling stack
```

Port                  | "9101"                                    | Application port

> AnnotationEndTime     | "annotations.scaling.exporter.time.end"   | Annotation end in HPA for create metrique
> AnnotationStartTime   | "annotations.scaling.exporter.time.start" | Annotation start in HPA for create metrique
> AnnotationMinReplicas | "annotations.scaling.exporter.replica.min"| Annotation in min HPA for create metrique

Reviewer: In English, if you use "for + verb", you must always add "ing" to the verb. Ex: for creating metric.

Anyway, I suggest:
```suggestion
AnnotationEndTime     | "annotations.scaling.exporter.time.end"   | Prescaling end time
AnnotationStartTime   | "annotations.scaling.exporter.time.start" | Prescaling start time
AnnotationMinReplicas | "annotations.scaling.exporter.replica.min"| Minimum of desired replicas during the prescaling
```

> LabelProject          | "project"                                 | label k8s where to retrieve the value for the project label of the metric

Reviewer: ```suggestion
LabelProject          | "project"                                 | k8s label used to add a label in the prescaling metric
```

# Prescaling Web server
## OpenAPI docs

> In the application is provided a swagger which is accessible with the endpoint. 
> 
> /swagger/index.html

Reviewer: ```suggestion
This application provides a swagger UI which is accessible on  `/swagger/index.html`.
```

## Event API 

> To allow the platform to scale up and down on different schedules and with a multiplier, pre-scaling events can be requested by the DRBs or the API.

Reviewer: ```suggestion
To allow the platform to scale up and down on different schedules and with a multiplier, prescaling events can be registered by the DRBs or the API.
```

DRBs?

> The API allows the creation/modification and deletion of Prescaling Event CRDs in the cluster
> 
> /api/v1/events/

Reviewer: ```suggestion
The following API allows the creation, modification and deletion of Prescaling Event CRDs in the cluster: `/api/v1/events/`.
```

## Metrics

> The metrics are exposed with the endpoint
> 
> /metrics

Reviewer: ```suggestion
The metrics are exposed on `/metrics` endpoint.
```
